common/logger/src/diem_logger.rs

```diff
@@ -239,7 +240,13 @@ impl DiemLoggerBuilder {
                 let mut filter_builder = Filter::builder();
 
                 if self.is_async && self.address.is_some() {
```

Comment:
The code looks good, but is there someway to test this?

Reply:
Also, is there a reason why the fallback to RUST_LOG if RUST_LOG_REMOTE isn't set?  Not sure why we need to do this.

Reply:
Tweaked a test to be able to ensure that the `RUST_LOG_REMOTE` env var is respected

Reply:
> Also, is there a reason why the fallback to RUST_LOG if RUST_LOG_REMOTE isn't set? Not sure why we need to do this.

??

Reply:
`RUST_LOG` is supposed to be the default variable for all logging.
`RUST_LOG_REMOTE` is the variable for remote logging.

We log two sources to local logs (e.g. pod logs) and to remote logs (e.g. vector / logstash / etc).

So, the fallback is if you set the variable for all the logs, you just get the same behavior everywhere

Reply:
Exactly, while in production we probably will always set both this is more about providing a sane default

Reply:
Got it, as long as that's well documented makes sense!